src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py

```diff
@@ -29,31 +29,60 @@ def test_lock_create_list_delete(self):
             self.cmd('az lock delete -n {}'.format(lock_name))
             self._sleep_for_lock_operation()
 
-    def test_readonly_lock_create_list_delete_resource_group(self):
-        self._lock_operation_with_resource_group('ReadOnly')
+    @ResourceGroupPreparer(name_prefix='cli_test_readonly_lock_create_list_delete_resource_group')
+    def test_readonly_lock_create_list_delete_resource_group(self, resource_group):
```

Comment:
Recommend rename: "test_readonly_resource_group_lock" for the method and prefix.